### ISSUE TITLE
Position.__eq__ method contains redundant code

### ISSUE THREAD

**Issue by btheado** [leo-editor/leo-editor#1470]
I came across this issue while writing code coverage tests for the comparison operators of the Position class (see https://groups.google.com/d/msg/leo-editor/Pq_IX9gFyb4/Rqk9xjylBQAJ for details). This seems like a very low priority issue and I might be misunderstanding the intent of the code. But if I do properly understand it, then the fix should be simple.

Here is a failing pytest:

```python
@pytest.fixture(scope="session")
def bridge():
    return leoBridge.controller(gui='nullGui',
        loadPlugins=False, readSettings=False,
        silent=False, verbose=False)

@pytest.fixture
def tree_1(bridge, tmpdir):
    """ Simple tree with just a single node """
    c = bridge.openLeoFile(str(tmpdir.join("tree_1.leo")))
    p = c.rootPosition()
    p.h = "1"
    p.b = "b_1"
    return c

def test_eq_subclass(tree_1):
    v = leoNodes.vnode(tree_1)
    class MyPosition(leoNodes.Position):
        pass
    p1 = leoNodes.Position(v=v)
    p2 = MyPosition(v=v)
    # I expected this to throw a NotImplementedError, but the code
    # is actually returning the class insted of throwing it. Bug? Seems
    # like it has to be a bug since python will try to cast the result to
    # a boolean which will work, but doesn't seem useful for anything.
    with pytest.raises(NotImplementedError):
        assert p2 == p1
```

The test fails with: `E           Failed: DID NOT RAISE <class 'NotImplementedError'>`

**Comment by edreamleo** [comment]
@btheado Thanks for running these coverage tests.  The code in question is:
```python
def __eq__(self, p2):
    """Return True if two positions are equivalent."""
    p1 = self
    # Don't use g.trace: it might call p.__eq__ or p.__ne__.
    if not isinstance(p2, Position):
        return False
    if p2 is None or p2.v is None:
        return p1.v is None
    if isinstance(p2, self.__class__):
        return (p1.v == p2.v and
            p1._childIndex == p2._childIndex and
            p1.stack == p2.stack)
    # Do this only after testing for None.
    return NotImplementedError
```
This is quite strange.  The test `if isinstance(p2, self.__class__):` seems to be redundant, since we already know that p2 is a Position.  That being so, the following should be correct:
```python
def __eq__(self, p2):
    """Return True if two positions are equivalent."""
    p1 = self
    # Don't use g.trace: it might call p.__eq__ or p.__ne__.
    if not isinstance(p2, Position):
        return False
    if p2 is None or p2.v is None:
        return p1.v is None
    return (p1.v == p2.v and
        p1._childIndex == p2._childIndex and
        p1.stack == p2.stack)
```

**Comment by edreamleo** [comment]
Any such change must be tested thoroughly, as early in the release cycle as possible.

**Comment by edreamleo** [comment]
Fixed at rev c7e3768 in devel.

**Comment by btheado** [comment]
I don't think the code was redundant. Try adding a trace before the return NotImplementedError and run my test and see that it gets executed. If the code were redundant, then the earlier return statement would prevent it from executing.  I was striving for code coverage and noticed that line wasn't being hit until I devised the test.

The class for p2 is **MyPostion** (which is a subclass of Position), not **Position**.

Maybe your fix is still fine, but it is not because the code is redundant. My guess about the original intent of the code was that if someone implemented a subclass of `Position`, then this would force them to override the `__eq__` method. If that guess is correct, then `NotIImplementedError` should be raised instead of returned.

**Comment by edreamleo** [comment]
@btheado Thanks for your comments.  Imo, the revised code is good enough.

I see no reason to ensure __eq__ is overridden in subclasses of Position.  In fact, I see no reason to subclass Position at all.  In any case, the subclass, not the base class, should be responsible for overriding __eq__, or any other method.
